unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/PoolManager/Pool.cs

```diff
@@ -113,6 +112,18 @@ private PoolableObject Extract()
             return po;
         }
 
+        public async UniTask PrewarmAsync(int prewarmCount, CancellationToken cancellationToken)
+        {
+            if (unusedObjects.Count >= prewarmCount)
+                return;
+                
+            for (int i = 0; i < prewarmCount; i++)
+            {
+                Instantiate();
+                await UniTask.NextFrame(cancellationToken);
+            }
```

Comment:
I would add a 
`cancellationToken?.ThrowIfCancelled` at the beginning of the method to avoid starting with a cancelled token.